### PR TITLE
Bigtable Read Xlang Wrapper for Python SDK

### PR DESCRIPTION
**Pull request by ahmedabu98**
Adding a Bigtable Read connector to Python, using the recently added [BigtableReadSchemaTransform](https://github.com/apache/beam/blob/master/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigTableReadSchemaTransformProvider.java).

### CODE REVIEW

**Comment by ahmedabu98** [comment]
Fixes #26643

**Comment by ahmedabu98** [comment]
Run  Python_Xlang_Gcp_Direct PostCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Direct PostCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Direct PostCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Dataflow PostCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Dataflow PostCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Dataflow PostCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Dataflow PostCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Direct PostCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Dataflow PostCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Dataflow PostCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Dataflow PostCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Dataflow PostCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Direct PostCommit

**Comment by ahmedabu98** [comment]
Run Java_Spark3_Versions PreCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Direct PostCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Dataflow PostCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Dataflow PostCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Direct PostCommit

**Comment by ahmedabu98** [comment]
Run Python_Xlang_Gcp_Direct PostCommit

**Comment by ahmedabu98** [comment]
R: @chamikaramj

**Comment by johnjcasey** [comment]
Run Java PreCommit

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableReadSchemaTransformProvider.java [hunk 1]
```diff
@@ -100,60 +100,55 @@ public List<String> outputCollectionNames() {
...
     @AutoValue.Builder
     public abstract static class Builder {
-      public abstract Builder setTable(String table);
-
-      public abstract Builder setInstance(String instance);
-
-      public abstract Builder setProject(String project);
-
-      abstract BigTableReadSchemaTransformConfiguration autoBuild();
+      public abstract Builder setTableId(String table);
```
Comment (chamikaramj) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableReadSchemaTransformProvider.java:130]: Nit: change variable names to match the setters (tableId, instanceId, projectId).

File: sdks/python/apache_beam/io/gcp/bigtableio.py [hunk 2]
```diff
@@ -227,3 +231,75 @@ def expand(self, pvalue):
...
+
+    return (
+        input.pipeline
+        | external_read
+        | beam.ParDo(self._BeamRowToPartialRowData()))
+
+  # PartialRowData has some useful methods for querying data within a row.
+  # To make use of those methods and to give Python users a more familiar
+  # object, we process each Beam Row and return a PartialRowData equivalent.
+  class _BeamRowToPartialRowData(beam.DoFn):
```
Comment (chamikaramj) [inline sdks/python/apache_beam/io/gcp/bigtableio.py:285]: Let's add unit tests to make sure that this conversion is correct for various data types etc.

File: sdks/python/apache_beam/io/gcp/bigtableio.py [hunk 3]
```diff
@@ -227,3 +231,75 @@ def expand(self, pvalue):
...
+      # initialize PartialRowData object
+      partial_row: PartialRowData = PartialRowData(key)
+      for fam_name, col_fam in families.items():
+        if fam_name not in partial_row.cells:
+          partial_row.cells[fam_name] = {}
+        for col_qualifier, cells in col_fam.items():
+          # store column qualifier as bytes to follow PartialRowData behavior
+          col_qualifier_bytes = col_qualifier.encode()
+          if col_qualifier not in partial_row.cells[fam_name]:
+            partial_row.cells[fam_name][col_qualifier_bytes] = []
```
Comment (chamikaramj) [inline sdks/python/apache_beam/io/gcp/bigtableio.py:299]: Also, let's add unit tests to cover the inner loop here.

**Review by chamikaramj — COMMENTED** [review]
Thanks.

File: sdks/python/apache_beam/io/gcp/bigtableio.py (hunk 2 above)
Comment (ahmedabu98) [reply sdks/python/apache_beam/io/gcp/bigtableio.py:285]: the data type returned from reading Bigtable cells is always bytes ([code documentation](https://cloud.google.com/java/docs/reference/google-cloud-bigtable/latest/com.google.bigtable.v2.Cell#com_google_bigtable_v2_Cell_getValue__))

File: sdks/python/apache_beam/io/gcp/bigtableio.py (hunk 3 above)
Comment (ahmedabu98) [reply sdks/python/apache_beam/io/gcp/bigtableio.py:299]: Added

**Comment by chamikaramj** [comment]
LGTM. Thanks.

**Comment by chamikaramj** [comment]
Run Java_GCP_IO_Direct PreCommit

**Comment by chamikaramj** [comment]
Run Java PreCommit

**Pull request merged**